pkg/controller/nodelifecycle/node_lifecycle_controller.go

```diff
@@ -431,6 +457,15 @@ func NewNodeLifecycleController(
 	nc.nodeLister = nodeInformer.Lister()
 	nc.nodeInformerSynced = nodeInformer.Informer().HasSynced
 
+	nc.podInformersSynced = func() bool {
```

Comment:
the original k8s PR #79370 doesn't have this code to enforce the podInformers synced. any reason we added here?